### ISSUE TITLE
move wad extraction to library

### ISSUE THREAD

**Issue by StarrFox** [vbe0201/kobold#14]
other people might want to use the extraction

**Comment by vbe0201** [comment]
I don't think this lends itself well to being a library function because it would be very inflexible.

`kobold_wad::Archive` currently enables random access by file paths and iterating over all files which should be good enough for people to work with.

**Comment by StarrFox** [comment]
https://tenor.com/view/lizard-dancing-lizard-cry-about-it-gecko-dance-gif-20175430
